libraries/dagster-iceberg/src/dagster_iceberg/io_manager/spark.py

```diff
@@ -37,8 +37,21 @@ def handle_output(
         """Writes a PySpark dataframe to an Iceberg table."""
         table_name = SparkIcebergDbClient.get_table_name(table_slice)
         table_exists = connection.catalog.tableExists(table_name)
+        writer = obj.writeTo(table_name)
         mode = "overwritePartitions" if table_exists else "create"
-        getattr(obj.writeTo(table_name), mode)()
+        if (
+            table_slice.partition_dimensions
+            and len(table_slice.partition_dimensions) > 0
+            and mode == "create"
+        ):
```

Comment:
I find this if statement hard to parse; it's really saying "if the table is intended to be partitioned and we're creating it". But then is len(table_slice.partition_dimensions) == 0 ever a valid state to be in? Should we just be erroring if we hit that?
Otherwise we'll just silently not partition it right?

Reply:
Ack. Seems fine given that we fix in a downstream